<HTML>
<HEAD>
<TITLE>System.FieldAccessException Class</TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF">
<H3>System.FieldAccessException Class</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public class FieldAccessException : MemberAccessException</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Base Types</H4>

<BLOCKQUOTE>
<A HREF="Object.html" TARGET="contents">Object</A><BR>
&nbsp;&nbsp;<A HREF="Exception.html" TARGET="contents">Exception</A><BR>
&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="SystemException.html" TARGET="contents">SystemException</A><BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="MemberAccessException.html" TARGET="contents">MemberAccessException</A><BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;FieldAccessException<P>

</BLOCKQUOTE>

<H4>Assembly</H4>

<BLOCKQUOTE>
mscorlib
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Represents the error that occurs when there is an attempt to access a field outside
      the scope in which access is permitted.<P>

</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
[<I>Note</I>: This exception is typically thrown when the access
      level of a field in a class library is changed, and
      one or more assemblies referencing the library have not been recompiled.]<P>

<P>

</BLOCKQUOTE>

<H4>Example</H4>

<BLOCKQUOTE>
The following example demonstrates a scenario under
      which <A HREF="FieldAccessException.html" TARGET="contents">FieldAccessException</A>
      is thrown.<P>

The following code contains a class with a public field (myField). This class
      is compiled into a class library.<P>

<PRE>using System;
namespace TestNameSpace 
{
 public class Class1
 {
   public Class1()
   { 
     Console.WriteLine (&quot;Constructing with public field&quot;);
   }
   public int myField = -1;
 }
}
   </PRE>
The following code references the class library above, and accesses
      TestNameSpace.Class1.myField. This code is compiled into an application.<P>

<PRE>using System;
using TestNameSpace;
class AppTest
{
  public static void Main()
  {
    Class1 test = new Class1();
    Console.WriteLine(&quot;Accessing member {0}.&quot;, test.myField);
  }
}
   </PRE>
The output of the application is<P>

<CODE> Constructing with public field<P>

Accessing member -1.<P>

</CODE>The code for the class library is changed and recompiled so that TestNameSpace.Class1.myField is no longer public. The following code changes
      myField from public to private. <P>

<PRE>using System;
namespace TestNameSpace 
{
 public class Class1
 {
   public Class1()
   {
     Console.WriteLine (&quot;Constructing with private field&quot;);
   }
   private int myField = -1;
 }
}
   </PRE>
When the application is executed again without being recompiled, the output
      is<P>

<CODE> Unhandled Exception: System.FieldAccessException:
         TestNameSpace.Class1.myField<P>

 at
         AppTest.Main()<P>

</CODE>
</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

<H4>Members</H4>

<BLOCKQUOTE>
<P>

FieldAccessException Constructors<P>

<A HREF="#FieldAccessException() Constructor" TARGET="contents">FieldAccessException() Constructor</A><BR>
<A HREF="#FieldAccessException(System.String) Constructor" TARGET="contents">FieldAccessException(System.String) Constructor</A><BR>
<A HREF="#FieldAccessException(System.String, System.Exception) Constructor" TARGET="contents">FieldAccessException(System.String, System.Exception) Constructor</A><BR>
</BLOCKQUOTE>

<HR>

<A NAME="FieldAccessException() Constructor"><H3>FieldAccessException() Constructor</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public FieldAccessException();</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Constructs and initializes a new instance of the
   <A HREF="FieldAccessException.html" TARGET="contents">FieldAccessException</A>
   class.<P>

</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
This constructor initializes the System.FieldAccessException.Message
property of the new instance to a system-supplied message that describes the
error, such as &quot;Attempted to access a private or protected field inside a type.&quot;
This message takes into account the current system culture.<P>

The System.FieldAccessException.InnerException property of the new instance
is initialized to <CODE>null</CODE>
 .<P>

</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="FieldAccessException.html" TARGET="contents">System.FieldAccessException Class</A>, <A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

</A><HR>

<A NAME="FieldAccessException(System.String) Constructor"><H3>FieldAccessException(System.String) Constructor</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public FieldAccessException(string message);</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Constructs and initializes a new instance of the
   <A HREF="FieldAccessException.html" TARGET="contents">FieldAccessException</A>
   class.<P>

</BLOCKQUOTE>

<H4>Parameters</H4>

<BLOCKQUOTE>
<DL>
<DT>message</DT>
<DD>A <A HREF="String.html" TARGET="contents">String</A> that describes the error. The content of <I>message</I> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</DD>
</DL>
</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
This constructor initializes the System.FieldAccessException.Message
property of the new instance using <I>message</I>. If <I>message</I> is
<CODE>null</CODE>
, the System.FieldAccessException.Message property is initialized to the 
system-supplied message provided by the constructor that takes no
arguments. <P>

 The System.FieldAccessException.InnerException property of
the new instance is initialized to <CODE>null</CODE>
.<P>

</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="FieldAccessException.html" TARGET="contents">System.FieldAccessException Class</A>, <A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

</A><HR>

<A NAME="FieldAccessException(System.String, System.Exception) Constructor"><H3>FieldAccessException(System.String, System.Exception) Constructor</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public FieldAccessException(string message, Exception inner);</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Constructs and initializes a new instance of the
   <A HREF="FieldAccessException.html" TARGET="contents">FieldAccessException</A> class.<P>

</BLOCKQUOTE>

<H4>Parameters</H4>

<BLOCKQUOTE>
<DL>
<DT>message</DT>
<DD>A <A HREF="String.html" TARGET="contents">String</A> that describes the error. The content of <I>message</I> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</DD>
<DT>inner</DT>
<DD>An instance of <A HREF="Exception.html" TARGET="contents">Exception</A> that is the cause of the current exception. If <I>inner</I> is not a <CODE>null</CODE>
 reference, the current exception was raised in a catch block handling <I>inner</I> .</DD>
</DL>
</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
This constructor initializes the System.FieldAccessException.Message property of the new instance
   using <I>message</I> and the System.FieldAccessException.InnerException property using
<I>inner</I>. If <I>message</I> is <CODE>null</CODE>
, 
   the System.FieldAccessException.Message property is initialized to the system-supplied message
   provided by the constructor that takes no arguments. <P>

[<I>Note</I>: For more information on inner
   exceptions, see System.Exception.InnerException.]<P>

</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="FieldAccessException.html" TARGET="contents">System.FieldAccessException Class</A>, <A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

</A></BODY>
</HTML>
